build_scripts/api/extensions/sd-aws-ext/scripts/api.py

```
@@ -0,0 +1,117 @@
+    async def settings():
+        max_models_num = external_code.get_max_models_num()
+        return {"control_net_max_models_num":max_models_num}
+
+    cached_cn_preprocessors = global_state.cache_preprocessors(global_state.cn_preprocessor_modules)
+    @app.post("/controlnet/detect")
+    async def detect(
+        controlnet_module: str = Body("none", title='Controlnet Module'),
+        controlnet_input_images: List[str] = Body([], title='Controlnet Input Images'),
+        controlnet_processor_res: int = Body(512, title='Controlnet Processor Resolution'),
+        controlnet_threshold_a: float = Body(64, title='Controlnet Threshold a'),
+        controlnet_threshold_b: float = Body(64, title='Controlnet Threshold b')
+    ):
+        controlnet_module = global_state.reverse_preprocessor_aliases.get(controlnet_module, controlnet_module)
+
+        if controlnet_module not in cached_cn_preprocessors:
+            raise HTTPException(
+                status_code=422, detail="Module not available")
+
+        if len(controlnet_input_images) == 0:
+            raise HTTPException(
+                status_code=422, detail="No image selected")
+
+        logger.info(f"Detecting {str(len(controlnet_input_images))} images with the {controlnet_module} module.")
+
+        results = []
+
+        processor_module = cached_cn_preprocessors[controlnet_module]
+
+        for input_image in controlnet_input_images:
+            img = external_code.to_base64_nparray(input_image)
+            results.append(processor_module(img, res=controlnet_processor_res, thr_a=controlnet_threshold_a, thr_b=controlnet_threshold_b)[0])
+
+        global_state.cn_preprocessor_unloadable.get(controlnet_module, lambda: None)()
+        results64 = list(map(encode_to_base64, results))
+        return {"images": results64, "info": "Success"}
+
+try:
+    import modules.script_callbacks as script_callbacks
+
+    script_callbacks.on_app_started(controlnet_api)
+except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```